### ISSUE TITLE
Use no_sync when doing gradient accumulation

### ISSUE THREAD

**Issue by achalddave** [mlfoundations/open_lm#48]
https://github.com/pytorch/pytorch/pull/72446

By default, FSDP will reduce gradients on every backward() call, which is slow in multi node settings. We should use fsdp.no_sync() to only reduce gradients on the last backward call.
